### ISSUE TITLE
Sorting a query adds in deleted documents

### ISSUE THREAD

**Issue by nirvdrum** [pubkey/rxdb#3631]
Apologies in advance if this is intended, but I'm seeing deleted documents appear in queries just by trying to sort the results. That is to say, the following query will not include deleted documents:

```ts
collection.find().exec()
```

But, adding a new sort to it will return deleted documents:

```ts
collection.find().sort({ rank: 'asc' })
```

I do not see the deleted documents with other query modifiers. E.g., adding an additional `where` clause does not include deleted documents:

```ts
collection.find().where('name').eq('Deleted Name')
```

For the time being, I can work around the issue by adding `where('_deleted').ne(true)` to any collection whose results I'd like to sort.

This is with RxDB 11.3.0.

**Comment by pubkey** [comment]
This is not intended. Looks like a bug.
Have you tried running the same query without the query builder?

```ts
collection.find({
  selector: {},
  sort: [{rank: 'ask'}]
})

```

**Comment by nirvdrum** [comment]
I see the same behavior without the query builder. Adding a sort option causes deleted documents to appear in the result.

**Comment by pubkey** [comment]
I assume this is with the lokijs storage? I will check.

**Comment by nirvdrum** [comment]
I'm using PouchDB with the "indexeddb" adapter. I can try with Loki as well.

**Comment by nirvdrum** [comment]
I'm sorry. This looks like a bug in pouchdb-adapter-indexeddb. Thank you for clarifying that adding new options to a query should not clear out the default ones. And thanks for confirming it works with loki-js. I tried switching to pouchdb-adapter-idb and ithe problem doesn't show up there. I debugged the calls and the same prepared query is being sent in both cases. I'll have to look into the adapter next.
